Refactor player widget and fix null safety issues

f8th opened the pull request:
This Pull Request addresses a critical issue where the app throws an "index out of bounds" exception when no songs are downloaded to the local file system. The exception occurred while building the player widget due to an unhandled null value in the song list.

![issue_snapshot](https://github.com/user-attachments/assets/753c5252-eae7-4f2e-9e28-b2a4153190c1)

Changes:
I added a null safety check to handle cases where no songs are available locally, preventing the app from crashing due to the index out-of-bounds error.

These changes ensure the app behaves gracefully when no songs are found, improving the user experience and preventing potential crashes.

f8th committed:
Refactor player widget and fix null safety issues

f8th committed:
Refactor AudioInfoWidget in player_view to handle null track values

rozpo commented on lib/app/views/player/player_view.dart:
```diff
@@ -90,19 +90,19 @@ class AudioInfoWidget extends StatelessWidget {
               ),
               Text(
-                track.album,
+                track?.artist ?? 'Unknown Artist',
```
Should be a `album` field not `artist`

rozpo replied:
Should be an `album` field not `artist`, same for text msg

rozpo commented on lib/app/widgets/player/player_widget.dart:
```diff
@@ -119,4 +119,4 @@ class ArtworkWidget extends StatelessWidget {
     );
   }
-}
+}
```
Missing new line at EOF

rozpo commented on lib/app/widgets/player/player_widget.dart:
```diff
@@ -67,15 +67,15 @@ class AudioInfoWidget extends StatelessWidget {
               Text(
-                track.title,
+                track?.title ?? 'Unknown Title',
               ),
-              Text(track.artist),
+              Text(track?.artist ?? 'Unknown Artist'),
```
Would you mind not use hardcoded msgs but instead add new entries for **artist**, **album** and **title** in `l10_en.arb` file, run `flutter gen-l10n` and then use them from `RetipL10n` mixin?

rozpo reviewed (commented):
Hey @Anonymous3017, thanks for your contribution to the project!

Please take a moment to review the comments I've added to your changes. Keep in mind that the app is still in its early stages of development, so there may be significant changes before the first release.

f8th committed:
Address code review feedback: use album field, localize strings, add newline at EOF

f8th commented:
Hey @rozpo  I have addressed all the requested changes:
- Updated the field to 'album' instead of 'artist'.
- Replaced hardcoded messages with localized strings from l10n.
- Added the missing newline at the end of the file.

Please review the updated code.
